radixdlt/src/integration/java/com/radixdlt/integration/recovery/RecoveryLivenessTest.java

```diff
@@ -196,46 +207,149 @@ private void restartNode(int index) {
 		}
 	}
 
+	private Timed<ControlledMessage> processNext() {
+		Timed<ControlledMessage> msg = this.network.nextMessage();
+		logger.debug("Processing message {}", msg);
+
+		int nodeIndex = msg.value().channelId().receiverIndex();
+		Injector injector = this.nodes.get(nodeIndex);
+		String bftNode = " " + injector.getInstance(Key.get(BFTNode.class, Self.class));
+		ThreadContext.put("bftNode", bftNode);
+		try {
+			injector.getInstance(DeterministicEpochsConsensusProcessor.class).handleMessage(msg.value().origin(), msg.value().message());
+		} finally {
+			ThreadContext.remove("bftNode");
+		}
+
+		return msg;
+	}
+
+	private Optional<EpochView> lastCommitViewEmitted() {
+		return network.allMessages().stream()
+			.filter(msg -> msg.message() instanceof EpochViewUpdate)
+			.map(msg -> (EpochViewUpdate) msg.message())
+			.map(e -> new EpochView(e.getEpoch(), e.getViewUpdate().getHighQC().highestCommittedQC().getView()))
+			.max(Comparator.naturalOrder());
+	}
+
+	private EpochView latestEpochView() {
+		return this.nodes.stream()
+			.map(i -> i.getInstance(Key.get(new TypeLiteral<DeterministicSavedLastEvent<EpochViewUpdate>>() { })).getLastEvent())
```

Comment:
The `Key.get(new TypeLiteral<DeterministicSavedLastEvent<EpochViewUpdate>>() { })` constant can be extracted.